algebird-core/src/main/scala/com/twitter/algebird/Preparer.scala

```diff
@@ -0,0 +1,132 @@
+package com.twitter.algebird
+
+trait Preparer[A, T, +This[A, T] <: Preparer[A, T, This]] extends HasMonoidAggregate[A, T] {
```

Comment:
So this is a one pass transformer: compose `A => TraversableOnce[B]` as many times as we like, then aggregate.

This kind of composition comes up so much, I wonder if we should have that as a thing by itself. These things form a category, and combine is `{ a => fn(a).flatMap(gn) }`